www/include/configuration/configObject/service_categories/DB-Func.php

```diff
@@ -84,56 +86,106 @@ function shouldNotBeEqTo0($value)
     }
 }
 
-function multipleServiceCategorieInDB($sc = array(), $nbrDup = array())
+function multipleServiceCategorieInDB($sc = [], $nbrDup = [])
 {
     global $pearDB, $centreon;
 
-    $scAcl = array();
+    $scAcl = [];
     foreach ($sc as $key => $value) {
-        $dbResult = $pearDB->query("SELECT * FROM `service_categories` WHERE `sc_id` = '" . $key . "' LIMIT 1");
-        $row = $dbResult->fetch();
-        $row["sc_id"] = null;
-        for ($i = 1; $i <= $nbrDup[$key]; $i++) {
+        $scId = filter_var($key, FILTER_VALIDATE_INT);
+        $query = "SELECT * FROM `service_categories` WHERE `sc_id` = :sc_id LIMIT 1";
+        $statement = $pearDB->prepare($query);
+        $statement->bindValue(':sc_id', $scId, \PDO::PARAM_INT);
+        $statement->execute();
+        $row = $statement->fetch();
+        for ($i = 1; $i <= $nbrDup[$scId]; $i++) {
             $val = null;
+            $bindParams = [];
             foreach ($row as $key2 => $value2) {
-                $key2 == "sc_name" ? ($sc_name = $value2 = $value2 . "_" . $i) : null;
-                $key2 == "sc_description" ? ($sc_alias = $value2 = $value2) : null;
+                switch ($key2) {
+                    case 'sc_name':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $sc_name = $value2 = $value2 . "_" . $i;
+                        $bindParams[':sc_name'] = [
+                            $value2 => \PDO::PARAM_STR
+                        ];
+                        break;
+                    case 'sc_description':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $bindParams[':sc_description'] = [
+                            $value2 => \PDO::PARAM_STR
+                        ];
+                        break;
+                    case 'level':
+                        $value2 = filter_var($value2 ?? false, FILTER_VALIDATE_INT);
+                        $value2
+                            ? $bindParams[':sc_level'] = [ $value2 => \PDO::PARAM_INT]
+                            : $bindParams[':sc_level'] = [ "NULL" => \PDO::PARAM_INT];
+                        break;
+                    case 'icon_id':
+                        $value2 = filter_var($value2 ?? false, FILTER_VALIDATE_INT);
+                        $value2
+                            ? $bindParams[':sc_icon_id'] = [ $value2 => \PDO::PARAM_INT]
+                            : $bindParams[':sc_icon_id'] = [ "NULL" => \PDO::PARAM_INT];
```

Comment:
I think this will not work
PDO will try to insert "NULL" string instead of null value

You may need to refactor a bit this bindParams array to do something like :
$bindParams[':sc_level'] = [\PDO::PARAM_INT => null];